libminifi/include/core/state/nodes/DeviceInformation.h

```diff
@@ -279,46 +278,46 @@ class Device {
     return macstr.length() > 0 ? std::to_string(hash_fn(macstr)) : "8675309";
   }
 #else
-std::string getDeviceId() {
-	PIP_ADAPTER_INFO adapterPtr;
-	PIP_ADAPTER_INFO adapter = NULL;
-
-	DWORD dwRetVal = 0;
-
-	std::hash<std::string> hash_fn;
-	std::set<std::string> macs;
-	
-	ULONG adapterLen = sizeof(IP_ADAPTER_INFO);
-	adapterPtr = (IP_ADAPTER_INFO *)malloc(sizeof(IP_ADAPTER_INFO));
-	if (adapterPtr == NULL) {
-		return "";
-	}
-	if (GetAdaptersInfo(adapterPtr, &adapterLen) == ERROR_BUFFER_OVERFLOW) {
-		free(adapterPtr);
-		adapterPtr = (IP_ADAPTER_INFO *)malloc(adapterLen);
-		if (adapterPtr == NULL) {
-			return "";
-		}
-	}
-
-	if ((dwRetVal = GetAdaptersInfo(adapterPtr, &adapterLen)) == NO_ERROR) {
-		adapter = adapterPtr;
-		while (adapter) {
-			char mac_add[13];
-			snprintf(mac_add, 13, "%02X%02X%02X%02X%02X%02X", adapter->Address[0], adapter->Address[1], adapter->Address[2], adapter->Address[3], adapter->Address[4], adapter->Address[5]);
-			macs.insert(mac_add);
-			adapter = adapter->Next;
-		}
-	}
-	
-	if (adapterPtr)
-		free(adapterPtr);
-	 std::string macstr;
+  std::string getDeviceId() {
+    PIP_ADAPTER_INFO adapterPtr;
+    PIP_ADAPTER_INFO adapter = NULL;
+
+    DWORD dwRetVal = 0;
+
+    std::hash<std::string> hash_fn;
+    std::set<std::string> macs;
+
+    ULONG adapterLen = sizeof(IP_ADAPTER_INFO);
+    adapterPtr = (IP_ADAPTER_INFO *)malloc(sizeof(IP_ADAPTER_INFO));
+    if (adapterPtr == NULL) {
+      return "";
+    }
+    if (GetAdaptersInfo(adapterPtr, &adapterLen) == ERROR_BUFFER_OVERFLOW) {
+      free(adapterPtr);
+      adapterPtr = (IP_ADAPTER_INFO *)malloc(adapterLen);
+      if (adapterPtr == NULL) {
+        return "";
+      }
+    }
+
+    if ((dwRetVal = GetAdaptersInfo(adapterPtr, &adapterLen)) == NO_ERROR) {
+      adapter = adapterPtr;
+      while (adapter) {
+        char mac_add[13];
+        snprintf(mac_add, 13, "%02X%02X%02X%02X%02X%02X", adapter->Address[0], adapter->Address[1], adapter->Address[2], adapter->Address[3], adapter->Address[4], adapter->Address[5]);
+        macs.insert(mac_add);
+        adapter = adapter->Next;
+      }
+    }
+
+    if (adapterPtr)
+    free(adapterPtr);
+    std::string macstr;
     for (auto &mac : macs) {
       macstr += mac;
```

Comment:
Do we just concat here without any separation?
Why doesn't this func simply return them in a vector?

Reply:
We don't need to.